libs/auth-api-lib/src/lib/services/delegations.service.ts

```diff
@@ -169,45 +184,76 @@ export class DelegationsService {
     })
 
     const filtered = result.filter(
-      (x) => x.delegationScopes !== null && x.delegationScopes!.length > 0,
+      (x) => x.delegationScopes != null && x.delegationScopes.length > 0,
     )
 
     return filtered.map((d) => d.toDTO())
   }
 
-  async create(
-    user: Auth,
+  private async getUserName(user: User) {
+    const response = await this.authFetch(
+      `${this.authConfig.issuer}/connect/userinfo`,
+      {
+        headers: {
+          Authorization: user.authorization,
+        },
+      },
+    )
+    const userinfo = (await response.json()) as { name: string }
+    return userinfo.name
+  }
+
+  private async getPersonName(
+    nationalId: string,
+    user: User,
     xRoadClient: string,
     authMiddlewareOptions: AuthMiddlewareOptions,
-    delegation: CreateDelegationDTO,
-  ): Promise<DelegationDTO | null> {
+  ) {
     const person = await this.personApi
       .withMiddleware(
         new AuthMiddleware(user, {
           forwardUserInfo: authMiddlewareOptions.forwardUserInfo,
         }),
       )
-      .einstaklingarGetEinstaklingur(<EinstaklingarGetEinstaklingurRequest>{
-        id: user.nationalId,
+      .einstaklingarGetEinstaklingur({
+        id: nationalId,
         xRoadClient: xRoadClient,
       })
-    if (!person || !user.nationalId) {
+    if (!person) {
       throw new BadRequestException(
-        `A person with nationalId<${user.nationalId}> could not be found`,
+        `A person with nationalId<${nationalId}> could not be found`,
       )
     }
+    return person.fulltNafn || person.nafn
+  }
+
+  async create(
+    user: User,
+    xRoadClient: string,
+    authMiddlewareOptions: AuthMiddlewareOptions,
+    delegation: CreateDelegationDTO,
+  ): Promise<DelegationDTO | null> {
+    const [fromDisplayName, toName] = await Promise.all([
+      this.getUserName(user),
+      this.getPersonName(
+        delegation.toNationalId,
+        user,
+        xRoadClient,
+        authMiddlewareOptions,
+      ),
+    ])
 
     this.logger.debug('Creating a new delegation')
     const id = uuid()
     await this.delegationModel.create({
       id: id,
       fromNationalId: user.nationalId,
       toNationalId: delegation.toNationalId,
-      fromDisplayName: person.fulltNafn || person.nafn,
-      toName: delegation.toName,
+      fromDisplayName,
+      toName,
```

Comment:
Should we not skip saving the names altogether. Just save the nationalIds and rely on the Identity resolver to fetch the correct names when appropriate?
Would fix cases where the user has e.g. changed their name etc.

Reply:
Great comment! That's absolutely the plan, but it depends on us getting a company registry api, implementing that fancy caching logic everywhere and performing name lookups in the IDS delegation picker.

This PR is a temporary improvement for the beta test.